packages/homeserver/src/middlewares/acl.middleware.ts

```diff
@@ -0,0 +1,63 @@
+import type { EventAuthorizationService } from '@hs/federation-sdk';
+
+const errCodes = {
+	M_UNAUTHORIZED: {
+		errcode: 'M_UNAUTHORIZED',
+		error: 'Invalid or missing signature',
+	},
+	M_FORBIDDEN: {
+		errcode: 'M_FORBIDDEN',
+		error: 'Access denied',
+	},
+	M_UNKNOWN: {
+		errcode: 'M_UNKNOWN',
+		error: 'Internal server error while processing request',
+	},
+};
+
+export const aclMiddleware = (federationAuth: EventAuthorizationService) => {
+	return async (context: any) => {
```

Comment:
I don tthink using any here is a good idea, it maybe a regular function/service and let each router implement its own middleware